src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -945,6 +965,12 @@ internal static RefKind GetModifiers(SyntaxTokenList modifiers, out SyntaxToken
                         Debug.Assert(refKind == RefKind.None);
                         isScoped = true;
                         break;
+                    case SyntaxKind.ReadOnlyKeyword:
+                        if (refKind == RefKind.Ref)
```

Comment:
>if (refKind == RefKind.Ref) [](http://example.com/codeflow?start=24&length=27)

It feels like the condition should be stricter to filter out various error situations. I think we should ignore `readonly` modifier, unless ```refKind == RefKind.Ref``` **and it is the token that immediately follows the ```refnessKeyword```**.